fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/SimplifyArithmeticRule.java

```diff
@@ -129,43 +135,68 @@ private static Expression process(BinaryArithmetic arithmetic, boolean isAddOrSu
 
     private static List<Operand> flatten(Expression expr, boolean isAddOrSub) {
         List<Operand> result = Lists.newArrayList();
-        if (isAddOrSub) {
-            flattenAddSubtract(true, expr, result);
-        } else {
-            flattenMultiplyDivide(true, expr, result);
-        }
+        doFlatten(true, expr, isAddOrSub, result, Optional.empty());
         return result;
     }
 
-    private static void flattenAddSubtract(boolean flag, Expression expr, List<Operand> result) {
-        if (TypeUtils.isAddOrSubtract(expr)) {
-            BinaryArithmetic arithmetic = (BinaryArithmetic) expr;
-            flattenAddSubtract(flag, arithmetic.left(), result);
-            if (TypeUtils.isSubtract(expr) && !flag) {
-                flattenAddSubtract(true, arithmetic.right(), result);
-            } else if (TypeUtils.isAdd(expr) && !flag) {
-                flattenAddSubtract(false, arithmetic.right(), result);
+    private static void doFlatten(boolean flag, Expression expr, boolean isAddOrSub, List<Operand> result,
```

Comment:
what does `flag` and `isAddOrSub` mean?
seams
`flag` means `expr` is positive when `flag == true` 
`isAddOrSub` means `expr` is add or sub
so, why not generate them in `doFatten`?

Reply:
SimplifyArithmeticRule.Operand.flag had annotation,  flag = true : + or *,     flag = false :  - or /

Reply:
isAddOrSub is the input argument.  it means only extract  +-  sub exprs,   or only extract */  sub exprs.